feat: Added QWidget

2b-t opened the pull request:
The disadvantage of the current code is that the widget itself is directly implemented  in the plugin. This way somebody can't embed the GUI into his/her own GUI easily (like in this tutorial [here](http://wiki.ros.org/rqt/Tutorials/Using%20.ui%20file%20in%20rqt%20plugin#Reusing_existing_GUI_class)). Therefore I have 
- **Separated the `Plugin` and the `QWidget`** (just like most of the other `rqt` packages do)
- Renamed the **files** in a similar fashion to the other packages with **underscores** instead of camelcased filenames. I made a simple example of how this can be useful [here](https://github.com/2b-t/rqt_embed_window_example) (see image below). Furthermore this eliminates the issue pointed out in #2 

![Screenshot](https://user-images.githubusercontent.com/53856473/123483960-e16c0500-d607-11eb-91f4-9d03cee64ca7.png)

awesomebytes commented on scripts/test_if_window_can_be_embedded.py:
```diff
@@ -48,6 +48,7 @@ def run_app(window_id):
 
     main_widget.show()
     app.exec_()
+    return
```
Is this return needed?

awesomebytes commented on src/rqt_embed_window/embed_window_widget.py:
```diff
@@ -31,7 +29,7 @@ def get_window_id_by_window_name(window_name):
             if str(this_window_name) == str(window_name):
                 return int(fields[0], 16)
-    return None
+    return
```
I believe that being explicit is better than being implicit, I'd prefer to leave the explicit return of `None` than to make it implicit.

awesomebytes commented on src/rqt_embed_window/embed_window_widget.py:
```diff
@@ -53,7 +51,7 @@ def get_window_id_by_pid(pid):
             if this_pid == pid:
                 return int(fields[0], 16)
-    return None
+    return
```
Same about being explicit preferred.

awesomebytes commented on src/rqt_embed_window/embed_window_widget.py:
```diff
@@ -136,55 +127,23 @@ def add_external_window_widget(self):
-    def shutdown_plugin(self):
-        # Free resources
-        self._process.kill()
+        return
```
I guess this return is unnecessary?

awesomebytes commented on src/rqt_embed_window/embed_window_widget.py:
```diff
@@ -136,55 +127,23 @@ def add_external_window_widget(self):
-        instance_settings.set_value("window_name", self._window_name)
+    def remove_widget(self):
+        self.verticalLayout.removeWidget(self._external_window_widget)
+        return
```
Same with this return being unnecessary?

awesomebytes commented on src/rqt_embed_window/embed_window_widget.py:
```diff
@@ -136,55 +127,23 @@ def add_external_window_widget(self):
+    
+    def kill_process(self):
+        self._process.kill()
+        return
```
Same with unnecessary return?

awesomebytes commented on src/rqt_embed_window/shell_cmd.py:
```diff
@@ -18,13 +18,15 @@ def __init__(self, cmd):
         self.process = subprocess.Popen(cmd, shell=True, stdin=self.inf,
                                         stdout=self.outf, stderr=self.errf,
                                         preexec_fn=os.setsid, close_fds=True)
+        return;
```
Unnecessary return?

awesomebytes commented on src/rqt_embed_window/shell_cmd.py:
```diff
@@ -18,13 +18,15 @@ def __init__(self, cmd):
         self.outf.close()
         self.errf.close()
         self.inf.close()
+        return;
```
Unnecessary return?

awesomebytes commented on src/rqt_embed_window/shell_cmd.py:
```diff
@@ -51,6 +53,7 @@ def is_succeeded(self):
     def wait_until_done(self):
         while not self.is_done():
             time.sleep(0.1)
+        return;
```
Unnecessary return?

awesomebytes commented on src/rqt_embed_window/shell_cmd.py:
```diff
@@ -60,6 +63,7 @@ def kill(self):
         self.retcode = -1
         os.killpg(self.process.pid, signal.SIGTERM)
         self.process.wait()
+        return;
```
Unnecessary return?

awesomebytes reviewed (changes requested):
Hey, first, let me apologize for taking forever to look at this. I switched jobs in between other things as you made the PR and forgot to look at it for a long while.

@2b-t  Thank you very much for doing this work. I would have liked to have it done as a QWidget, but I did not get there!

I have a few comments in this PR which are basically minor in regards of `return`s that are either unnecessary (and I had some of them in my code already!) or that I'd prefer to have an explicit return of `None` to help other possible contributors understand what is going on by being explicit (this is a personal preference, I'm open to discussion as I may be missing something).

I'm keen to merge this. If you have the time to attend my comments, that would be great.

There's one thing that would worry me a bit, which is that the plugin name has changed from `RqtEmbedWindowWidget` to `EmbedWindowWidget` (which makes total sense) and that could break people's code that points to the old name. I wonder if we can do a wrapper for the old plugin name that just calls the new one, so when merging this into main we won't be breaking anyone's code?
Maybe it's a lost cause and it would break people's rqt configs anyways.

Again, thanks a lot, and I apologize for not merging straight away.

2b-t commented:
Hey @awesomebytes
Thanks for your reply. No worries, I completely understand!
Coming from C++ I am used to `return;` but indeed `return None` might be more expressive and using no return at all in the initializer might be more pythonic. 
Regarding the re-naming: I only tried to keep the nomenclature consistent to other Python rqt plug-ins. I think it should be possible to wrap it or simply alias it to guarantee backwards compatibility. I will have a look at it tomorrow!

2b-t commented:
@awesomebytes I just **removed all unnecessary returns** and made the others **explicitly `return None`**. Furthermore I **fixed a small bug** that I introduced when creating the separate widget in the beginning that would show 
```
PluginHandlerDirect._shutdown_plugin() plugin "rqt_embed_window/EmbedWindow#1" raised an exception:
Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python2.7/dist-packages/qt_gui/plugin_handler_direct.py", line 87, in _shutdown_plugin
    self._plugin.shutdown_plugin()
  File "/home/user/catkin_ws/src/rqt_embed_window/src/rqt_embed_window/embed_window.py", line 40, in shutdown_plugin
    self._widget.kill_process()
  File "/home/user/catkin_ws/src/rqt_embed_window/src/rqt_embed_window/embed_window_widget.py", line 146, in kill_process
    if process:
NameError: global name 'process' is not defined
```
in case no process was added to the widget and the widget was closed.

___

I also checked back on the **naming** of the widget and the file that I changed from camel case to underscores removing the leading `Rqt`. The reason why I renamed it was for it to be **consistent with the other official rqt packages** like [Rqt Publisher](https://github.com/ros-visualization/rqt_publisher/tree/master/src/rqt_publisher) or [Rqt Topic](https://github.com/ros-visualization/rqt_topic/tree/master/src/rqt_topic).
In my opinion the changes to the naming are not strictly relevant to the end-user as they not the intended public API: One would use the plugin by either launching it from `rosrun rqt_gui rqt_gui` or by launching it directly with `rosrun rqt_embed_window rqt_embed_window`. Both of these remain unaffected by the change.

The changes are only relevant in case somebody would refer to **`RqtEmbedWindow.py`** explicitly either by
- executing  `rosrun rqt_embed_window RqtEmbedWindow.py` which should result in an error anyhow or 
- **including the file into their project** which would only make sense if they would want to inherit from it in order to derive their own plugin.

In case you are worried that somebody might have done latter we could add a file called `RqtEmbedWindow.py` that imports `embed_window.py` with `from .embed_window import EmbedWindow as RqtEmbedWindow` (or alternatively imports it and defines an alias for the class `RqtEmbedWindow = EmbedWindow`) as the two plugins still share the exact same interface. If one would have touched the internals of the class though, their code might break in any case due to the introduction of the widget.

Taking these things into consideration I think this is a very unlikely scenario and I would still keep the changes in naming in order to be consistent with the other rqt packages.

2b-t commented:
Hey @awesomebytes 
Have you any updates on this?

2b-t commented:
_Hi @awesomebytes_ 
Been a while now. Could we finally merge this?
I would also be interested in porting this to ROS 2 Humble and making another pull request for this at some point...
